smeargle.py

```diff
@@ -177,103 +184,109 @@ def sd_api_call():
-                            sd_api_call()
-                        elif bool(dry_run_enabled) == True:
-                            api_call_count += 1
-                        break
-                    elif bool(regenerate_ai_artwork) == False:
-                        print("cover.png detected to be ai generated, but regenerate_ai_artwork is set to false. Skipping\n")
-                        break
-            else:
-                print("cover art does not exist. calling stable diffusion api")
-                print(f"sending sd api call:\ntitle: {title_item}\nartist: {artist_item}")
-                if bool(dry_run_enabled) == False:
-                    sd_api_call()
-                elif bool(dry_run_enabled) == True:
-                    api_call_count += 1
-                print(f"Total images created: {api_call_count}")
-                print("changing directory\n")
-                break
-        elif musicfile.endswith(".opus"):
-            musicfilepath = os.path.join(dirpath, musicfile)
-            print(f".opus found: {musicfilepath}")
-            print(f"directory: {dirpath}")
-            f = music_tag.load_file(f"{musicfilepath}")
-            title_item = f['title']
-            artist_item = f['artist']
-            # check if cover.png exists (common artwork format)
-            if os.path.isfile(f"{dirpath}\cover.png"):
-                print("cover art exists (cover.png). checking if ai generated")
-                im = Image.open(f'{dirpath}\cover.png')
-                im.load()
-                # check if cover.png's author is "AI" (signature of images generated by this script)
-                if im.info['Author'] == str("AI"):
-                    if bool(regenerate_ai_artwork) == True:
-                        print("cover.png detected to be ai generated. regenerating")
-                        if bool(dry_run_enabled) == False:
-                            sd_api_call()
-                        elif bool(dry_run_enabled) == True:
-                            api_call_count += 1
-                        break
-                    elif bool(regenerate_ai_artwork) == False:
-                        print("cover.png detected to be ai generated, but regenerate_ai_artwork is set to false. Skipping\n")
-                        break
-            # check if AlbumArt.jpg exists (common artwork format)
-            elif os.path.isfile(f"{dirpath}\AlbumArt.jpg"):
-                print("cover art exists (AlbumArt.jpg). skipping this file\n")
-                break
+
+def check_author_ai(dirpath, name:str = "AI"):
+    im = Image.open(dirpath)
+    im.load()
+    return im.info['Author'] == name
+
+def check_and_generate(dirpath, musicfile, music_extension):
+    musicfilepath = os.path.join(dirpath, musicfile)
+    print(f"{music_extension} found: {musicfilepath}")
+    
+    tag = music_tag.load_file(f"{musicfilepath}")
+
+    # check if there is any artwork attached to the .mp3 file ("artwork" tag)

```

Comment:
As far as I could tell, the only difference between .mp3 and opus was this extra check for tag['artwork']

Reply:
Correct. The extra check for artwork tags is done differently between the two file types. mp3 is very easy to detect artwork as it is an easily readable tag in the file that has had wide support for a long time. But opus is different, and more complicated (see https://wiki.xiph.org/VorbisComment#METADATA_BLOCK_PICTURE). The original goal for .opus was to encode the stable diffusion image into the proper format for adding to the METADATA_BLOCK_PICTURE tag, but I wasn't able to get it working well enough to publish. 

I decided to skip the typical opus artwork check in favor or just checking if smeargle AI artwork already exists in the form of cover.png. 

Good fix, will include.